app/controllers/withdraw_controller.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+class WithdrawController < ApplicationController
+  def create
+    @document = Document.with_current_edition.find_by_param(params[:id])
+    edition = @document.current_edition
+    @public_explanation =
+      edition.withdrawn? ? edition.status&.details&.public_explanation : nil
+
+    if !current_user.has_permission?(User::PRE_RELEASE_FEATURES_PERMISSION)
+      render :withdraw
+      return
+    end
+
+    if current_user.has_permission?(User::MANAGING_EDITOR_PERMISSION)
+      render :create
+    else
+      render :non_managing_editor, status: :forbidden
+    end
+  end
+
+  def new
+    unless user_has_permissions?
+      # FIXME: this shouldn't be an exception but we've not worked out the
+      # right response - maybe bad request or a redirect with flash?
+      raise "Can't withdraw an edition without permissions"
+    end
+
+    Document.transaction do
+      @document = Document.with_current_edition.lock!.find_by_param(params[:id])
+      public_explanation = params[:public_explanation]
+      issues = Requirements::WithdrawalChecker.new(public_explanation).pre_withdrawal_issues
+
+      if issues.any?
+        flash["alert_with_items"] = {
+          "title" => I18n.t!("withdraw.create.flashes.requirements"),
+          "items" => issues.items,
+        }
+
+        render :create
+      else
+        begin
+          raise "Cannot withdraw an edition that is not live" unless withdrawable?(@document)
+
+          UnpublishService.new.withdraw(@document.current_edition, public_explanation, current_user)
+        rescue GdsApi::BaseError => e
+          GovukError.notify(e)
+          redirect_to withdraw_path,
+            alert_with_description: t("withdraw.create.flashes.publishing_api_error"),
+            public_explanation: public_explanation
+          return
+        end
+        redirect_to @document
```

Comment:
super minor but might want to have this inside the begin end then you can skip the return inside the rescue.

Reply:
I can't do that because I get a Double Render error.

Reply:
I assume you get double render error if the redirect_to isn't moved to inside the begin?